executor/src/witgen/machines/block_machine.rs

```diff
@@ -326,14 +372,17 @@ impl<T: FieldElement> BlockMachine<T> {
                             })
```

Comment:
Can you avoid cloning the name above (I know it's not your code)?
Why do we need a name in the first place? Maybe we can just pass `PolynomialReference`s?